### PR TITLE
Picklable action spaces

### PR DESCRIPTION
**Pull request by shwang**
Before this PR, any `MineRLEnv.action_space` was unpicklable, causing errors when saving some Stable Baselines policies. The unpicklablilty was due to `MineRLEnv.action_space.noop` closing over the `MineRLEnv` instance (which we wouldn't reasonably expect to be serializable).

AFAICT `action_space.noop`, which is dynamically added to MineRLEnv.action_space upon initialization of MineRLEnv, is functionally identical (except in one case) to `action_space.no_op`, which unlike `noop` is picklable because it doesn't close over the entire Env.

The one case in which `noop` is different from `no_op` is when `MineRLEnv.__init__(...)` receives an optional `noop_action` argument, in which case `noop` always returns that argument. It doesn't look like any code path sets the `noop_action` argument (from searching "noop_action").

There's also a third way to access the no-op action: the `MineRLEnv.noop_action(self)` method.

After this PR, the only way to access the no-op action is the allow-action-spaces-to-be-picklable variant, which I have renamed from `no_op()` to `noop()` because the MineRL docs and code most often uses `noop()` instead of `no_op()`. I've found and replaced references to `no_op()` and `noop_action()`.

I also deleted the `MineRLEnv.__init__` method's `noop_action` argument and related code, as it is now unused and I think it's unlikely that we will want custom no_op actions. (I don't think it would be hard to get the `noop_action` feature this working and picklable right now if anyone objects to this removal.)

### CODE REVIEW

File: minerl/env/core.py [hunk 1]
```diff
@@ -175,31 +171,6 @@ def _robust_launch_new_instance(self, *, max_tries=3) -> InstanceManager:
         raise RuntimeError(f"Failed to build and launch Minecraft instance "
                            f"{max_tries} times. Giving up.")
 
-    def _setup_spaces(self, observation_space, action_space):
-        self.action_space = action_space
-        self.observation_space = observation_space
-
-        def map_space(space):
```
Comment (decodyng) [inline minerl/env/core.py:182]: I'm a little confused about what this `map_space` function was doing; was it just duplicating the no_op action for each space? If so, why does it manually specify a value rather than just calling the no_op function directly? I just want to confirm this wasn't doing something subtle we're not realizing as valuable before we delete it.

**Review by decodyng — COMMENTED** [review]
Mostly looks good, though it's a little bit hard to follow the musical chairs of which function is going where, and I want to confirm that we're not losing intended functionality with the removal of _setup_spaces

File: minerl/env/core.py (hunk 1 above)
Comment (shwang) [reply minerl/env/core.py:182]: Double-checking this seems like a good idea.

I'm 95% sure that `map_space` just reimplements what is currently called [the `.no_op` method ](https://github.com/HumanCompatibleAI/minerl/blob/master/minerl/env/spaces.py#L42)in every custom Space, and that the result of `map_space` is saved as `minerl_env._default_action` to be returned via the dynamically set `minerl_env.action_space.noop()` action (which has to close over `minerl_env` in order to return `minerl_env._default_action`).

Comment (shwang) [reply minerl/env/core.py:182]: I believe the engineering reason behind having `map_space` save to `minerl_env._default_action` and have `noop()` return a deepcopy of `minerl_env._default_action` is so that at initialization that MineRLEnv can have a custom noop action via the `noop_action` parameter.

Comment (shwang) [reply minerl/env/core.py:182]: (Nothing uses this custom noop action right now AFAICT with Pycharm searching for `noop_action`)

Comment (shwang) [reply minerl/env/core.py:182]: Oh, this is prob relevant for understanding -- we only ever call `map_space` if the custom noop action wasn't set https://github.com/HumanCompatibleAI/minerl/blob/master/minerl/env/core.py#L193

Comment (decodyng) [reply minerl/env/core.py:182]: Yeah, I think my question is something like: if this is the same as no_op in every space, why don't they just assign the result from the space's no_op() function  to `_default_action` if there is no default action passed in?

Comment (shwang) [reply minerl/env/core.py:182]: I don't know.

Maybe the author forgot that there was the other implementation before they coded this in? Or maybe `map_spaces` is a copy of some original implementation from Malmo, and the MineRL gym spaces are the new implementation. There's an [unreached line](https://github.com/HumanCompatibleAI/minerl/blob/master/minerl/env/core.py#L189) in `map_spaces` referencing a method that doesn't seem to exist in any MineRL space or standard Gym space, which seems to support the "copied from somewhere else" hypothesis.

Comment (decodyng) [reply minerl/env/core.py:182]: 🤦‍♀ Yeah, a frustrating thing about MineRL being occasionally such a kludge of a codebase (see also: an entire top level directory of source that was just never called, but left confusingly lying around) is that it's often hard to know how hard to chesterton's fence on it. 

Could we do a quick empirical check by temporarily bringing back the `map_space` code base and ensuring the no_ops are in fact empirical for some set of action spaces and observation spaces in existing MineRL environments?

Comment (shwang) [reply minerl/env/core.py:182]: Sure, I'll use the current `master` to compare outputs from `no_op` and `noop`.

Comment (shwang) [reply minerl/env/core.py:182]: Confirmed that `no_op` and `noop` are the same for four arbitrary envs (containing Box, Enum, and Discrete action spaces).

https://gist.github.com/shwang/87176375e62e71dad0488f1ce1b00fca

Comment (decodyng) [reply minerl/env/core.py:182]: Awesome, thanks for doing that.

**Review by decodyng — APPROVED** [review]
LGTM given equivalence testing
